torch/_inductor/codegen/triton.py

```diff
@@ -1171,11 +1173,28 @@ def __init__(self, line, var, mask, size_map):
                 self.size_map = size_map
 
             def __call__(self):
-                # The conditions need to be in parens because of Python's operator precedence.
-                # It'd be less # error-prone to use and/or/not, which is suported by triton
-                size = self.size_map[(self.var, self.mask)]
-                cond = f"(0 <= {self.var}) & ({self.var} < {size})"
-                cond_print = f"0 <= {self.var} < {size}"
+                size, size_str = self.size_map[(self.var, self.mask)]
+
+                # We assert if we've not been able to prove the bound
+                assert_min = not (self.var.bounds.lower >= 0)
+                assert_max = not (self.var.bounds.upper < size)
```

Comment:
How does this work? If I try this locally with a sympy Expression I get
```
TypeError: cannot determine truth value of Relational
```

Reply:
Yep, I thought it was equivalent but it's not even equivalent. Now I argue that the two previous options were wrong. Now I think I got the condition right.